x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java

```diff
@@ -129,8 +129,9 @@ protected void doLookupUser(String username, ActionListener<User> listener) {
         if (realmEnabled == false) {
             if (anonymousEnabled && AnonymousUser.isAnonymousUsername(username, config.settings())) {
                 listener.onResponse(anonymousUser);
+            } else {
+                listener.onResponse(null);
```

Comment:
can you add a test to ensure the listener is only called once?

Reply:
yes, good call, will do

Reply:
I recently had to do exactly that, if you're looking for a possible approach, look here: https://github.com/elastic/elasticsearch/pull/60497

Reply:
added in ad62802